data/data/gcp/cluster/dns/base.tf

```diff
@@ -18,27 +20,32 @@ resource "google_dns_managed_zone" "int" {
 }
 
 resource "google_dns_record_set" "api_external" {
-  count = var.public_endpoints ? 1 : 0
+  count = var.public_endpoints && var.create_public_zone_records ? 1 : 0
 
   name         = "api.${var.cluster_domain}."
   type         = "A"
   ttl          = "60"
   managed_zone = var.public_dns_zone_name
+  project      = var.network_project_id == "" ? var.project_id : var.network_project_id
```

Comment:
FYI... when `project` is set to an empty string, it will default to the provider specified project.... so using empty strings to fall through to the "service project" will work when public_dns_zone_project and/or private_dns_zone_project are not specified.